wasmtime-api/src/externals.rs

```diff
@@ -234,36 +242,142 @@ impl Global {
 }
 
 pub struct Table {
-    _store: Rc<RefCell<Store>>,
+    store: Rc<RefCell<Store>>,
     r#type: TableType,
+    wasmtime_handle: InstanceHandle,
+    wasmtime_export: wasmtime_runtime::Export,
+}
+
+fn get_table_item(
+    handle: &InstanceHandle,
+    store: &Rc<RefCell<Store>>,
+    table_index: cranelift_wasm::DefinedTableIndex,
+    item_index: u32,
+) -> Val {
+    if let Some(item) = handle.table_get(table_index, item_index) {
+        from_checked_anyfunc(item, store)
+    } else {
+        AnyRef::null().into()
+    }
+}
+
+fn set_table_item(
+    handle: &mut InstanceHandle,
+    store: &Rc<RefCell<Store>>,
+    table_index: cranelift_wasm::DefinedTableIndex,
+    item_index: u32,
+    val: Val,
+) -> bool {
+    let item = into_checked_anyfunc(val, store);
+    if let Some(item_ref) = handle.table_get_mut(table_index, item_index) {
+        *item_ref = item;
+        true
+    } else {
+        false
+    }
 }
 
 impl Table {
-    pub fn new(store: Rc<RefCell<Store>>, r#type: TableType) -> Table {
+    pub fn new(store: Rc<RefCell<Store>>, r#type: TableType, init: Val) -> Table {
+        match r#type.element() {
+            ValType::FuncRef => (),
+            _ => panic!("table is not for funcref"),
+        }
+        let (mut wasmtime_handle, wasmtime_export) =
+            generate_table_export(&r#type).expect("generated table");
+
+        // Initialize entries with the init value.
+        match wasmtime_export {
+            wasmtime_runtime::Export::Table { definition, .. } => {
+                let index = wasmtime_handle.table_index(unsafe { &*definition });
+                let len = unsafe { (*definition).current_elements as u32 };
```

Comment:
Would it make sense to change `current_elements` itself to a `u32`, so that we don't have to do a narrowing cast here?